sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,323 @@
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+
+        """
+        self._scorer = check_scoring(self.estimator, scoring=self.scoring)
+
+        if not isinstance(self.n_features_to_select, int) and\
+                not isinstance(self.n_features_to_select, tuple):
+            raise ValueError('n_features_to_select must be a positive integer'
+                             ' or tuple')
+
+        if isinstance(self.n_features_to_select, int) and (
+                    self.n_features_to_select < 1 or
+                    self.n_features_to_select > X.shape[1]):
+            raise ValueError('n_features_to_select must be a positive integer'
+                             ' between 1 and X.shape[1], got %s'
+                             % (self.n_features_to_select, ))
+
+        if isinstance(self.n_features_to_select, tuple):
+            if len(self.n_features_to_select) != 2:
+                raise ValueError('n_features_to_select tuple must consist of 2'
+                                 ' elements a min and a max value.')
+
+            if self.n_features_to_select[0] not in range(1, X.shape[1] + 1):
+                raise ValueError('n_features_to_select tuple min value must'
+                                 ' be in range(1, X.shape[1]+1).')
+
+            if self.n_features_to_select[1] not in range(1, X.shape[1] + 1):
+                raise ValueError('n_features_to_select tuple max value must'
+                                 ' be in range(1, X.shape[1]+1).')
+
+            if self.n_features_to_select[0] > self.n_features_to_select[1]:
+                raise ValueError('The min n_features_to_select value must be'
+                                 ' smaller than the max'
+                                 ' n_features_to_select value.')
+
+        if isinstance(self.n_features_to_select, tuple):
+            select_in_range = True
+        else:
+            select_in_range = False
+            k_to_select = self.n_features_to_select
+
+        self.estimator_ = clone(self.estimator)
+
+        self._n_features = X.shape[1]
+        self.subsets_ = {}
+        orig_set = set(range(self._n_features))
+        if self.forward:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[1]
+            k_idx = ()
+            k = 0
+        else:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[0]
+            k_idx = tuple(range(self._n_features))
+            k = len(k_idx)
+            k_score = self._calc_score(X, y, k_idx)
+            self.subsets_[k] = {
+                'feature_subset_idx': k_idx,
+                'cv_scores': k_score,
+                'avg_score': np.nanmean(k_score)
```

Comment:
Why nanmean?

Reply:
This is in case some scorers might return "nan's" (I think this can happen for regressors in certain edge cases). Happy to change this to a regular np.mean() -- what do you think?

Reply:
I'd think for scorers returning nans, it would be dishonest and inappropriate to just count everything else...